sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -709,10 +697,8 @@ public static FileOperations getFileOperations(String spec, PipelineOptions opti
      */
      void copy(List<String> srcFilenames, List<String> destFilenames) throws IOException;
 
-    /**
-     * Remove a collection of files.
-     */
-    void remove(Collection<String> filenames) throws IOException;
+    /** Removes a directory and the files in it (but not subdirectories). */
+    void removeDirectoryAndFiles(String directory) throws IOException;
```

Comment:
This seems like it's going in the opposite direction of @peihe 's proposal for the new API. I'm fine with it, but surprised that @peihe signed off given that he'll just be rewriting this soon.